src/socket.ts

```diff
@@ -133,12 +136,16 @@ export class ConstellationSocket extends EventEmitter {
         });
 
         this.on('close', err => {
+            if (this.state === State.Reconnecting) {
+                return;
+            }
```

Comment:
I think there was some misunderstanding talking about states here. The `Reconnecting` state was not a good name, let's call it `Refreshing`. My idea was that:

 - if `connect()` is called while a socket is in the `Closing` state, it should transition to the `Refreshing` state and return
 - when a socket `close` event is fired when we're `Refreshing`, we should transition to the `Idle` state and immedately call connect()
 - when we're waiting to connect down on L148, the state would be in a `Reconnecting` state
 - `.close()` should see that if we are in a `Reconnecting` state, the reconnectTimeout should be cancelled and we should transition immediate. Otherwise it should be put into the `Closing` state.  (the remediates a previous oversight on my part which would cause the socket never to enter a Closed state if .close is called while the reconnectTimeout is running)

At least, this was my thought. State-based logic is sometimes a bear to deal with :/